### ISSUE TITLE
Add median duration markline to dag run duration page

### ISSUE THREAD

**Issue by tirkarthi** [apache/airflow#39667]
### Description

Task duration chart already has median run and queued duration values as markline using echarts. However dag run duration had issues implementing this since the show landing pages is a checkbox that dynamically adds another series and due to an issue in echarts the markline was not rendering properly. Using clear method works but it loses the animations and reduces UX. Ref comments.

Since legends are clickable to show and hide corresponding series a proposed solution would be to have landing duration added by default along with legend but have it disabled by default. Users who need it can click on the legend to enable it and the series will be rendered. This is similar to using a checkbox except using legends and the selection can be saved to persist between pages if there are no issues like the dynamic series update as above.

https://github.com/apache/airflow/pull/38434#issuecomment-2016693194
https://github.com/apache/airflow/pull/38434#issuecomment-2016721461

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by tirkarthi** [comment]
I tried an attempt at this to make this similar to task duration page but in many cases like catchup the landing page time is in days so days is applied as a unit and the actual runs which are complete in the order of minutes will be displayed on the scale of days which is not good UX. The primary issue is that landing times and dag run duration would have different units and to scale them based on selection which is not the case with task duration since the run duration is the primary field for unit with queued duration mostly in seconds to minutes.

**Comment by eladkal** [comment]
@tirkarthi is this handled in Airflow 3?

**Comment by tirkarthi** [comment]
No, the duration charts are not implemented for tasks and dagruns in Airflow 3. There is still some work that needs to be done in the API end to get the duration data as per design i.e. last 14 runs, grouped by success/failed rate, grouped by date.

cc: @pierrejeambrun @bbovenzi

**Comment by pierrejeambrun** [comment]
Checking at airflow 3 UI, we do have a median mark for dag run duration chart:

<img width="1047" height="475" alt="Image" src="https://github.com/user-attachments/assets/24170d35-cdb5-453c-9041-b8bc75afb295" />

**Comment by pierrejeambrun** [comment]
Marking as completed, feel free to re-open if I missed something.
